From saeed@marvell.com  Tue Dec 22 11:50:16 2009
From: Saeed Bishara <saeed@marvell.com>
Date: Mon, 14 Dec 2009 17:17:33 +0200
Subject: USB: ehci: add call of free_cached_itd_list() function in disable_periodic()
To: linux-usb@vger.kernel.org
Cc: Dmitri Epshtein <dima@marvell.com>, Saeed Bishara <saeed@marvell.com>
Message-ID: <1260803854-4606-1-git-send-email-saeed@marvell.com>


From: Dmitri Epshtein <dima@marvell.com>

Sometimes disable_periodic() stop scan_periodic before than free_cached_itd_list() was called.
In such case USB Host stacked during disconnect operation
Solution: add call of free_cached_itd_list() function in disable_periodic()

Signed-off-by: Dimitry Epshtein <dima@marvell.com>
Signed-off-by: Saeed Bishara <saeed@marvell.com>
Cc: David Brownell <dbrownell@users.sourceforge.net>
Cc: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ehci-sched.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/host/ehci-sched.c
+++ b/drivers/usb/host/ehci-sched.c
@@ -510,6 +510,8 @@ static int disable_periodic (struct ehci
 	ehci_writel(ehci, cmd, &ehci->regs->command);
 	/* posted write ... */
 
+	free_cached_itd_list(ehci);
+
 	ehci->next_uframe = -1;
 	return 0;
 }
